src/App.js

```diff
@@ -1,15 +1,19 @@
 import { Route, Routes } from 'react-router-dom';
-import { Archives, Notes, Profile, Trash } from './pages/index';
-import { LandingPage } from "./components";
+import { Archives, Login, Notes, Profile, Signup, Trash } from './pages/index';
+import { LandingPage, RequireAuth } from "./components";
 function App() {
   return (
     <>
       <Routes>
         <Route path="/" element={<LandingPage />} />
-        <Route path="/profile" element={<Profile />} />
-        <Route path="/notes" element={<Notes />} />
-        <Route path="/archives" element={<Archives />} />
-        <Route path="/trash" element={<Trash />} />
+        <Route path="/signup" element={<Signup />} />
+        <Route path="/login" element={<Login />} />
+
+
+        <Route path="/profile" element={<RequireAuth> <Profile /></RequireAuth>} />
+        <Route path="/notes" element={<RequireAuth> <Notes /></RequireAuth>} />
+        <Route path="/archives" element={<RequireAuth> <Archives /></RequireAuth>} />
+        <Route path="/trash" element={<RequireAuth> <Trash /></RequireAuth>} />
       </Routes>
```

Comment:
suggestion: you can use 'outlet' feature provided by react-router for protecting private routes.

Reply:
Noted, will look into that.